nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CaptureChangeMySQL.java

```diff
@@ -0,0 +1,928 @@
+    private final Serializer<TableInfo> cacheValueSerializer = new TableInfo.Serializer();
+    private final Deserializer<TableInfo> cacheValueDeserializer = new TableInfo.Deserializer();
+
+    private Connection jdbcConnection = null;
+
+    private static final BeginTransactionEventWriter beginEventWriter = new BeginTransactionEventWriter();
+    private static final CommitTransactionEventWriter commitEventWriter = new CommitTransactionEventWriter();
+    private static final SchemaChangeEventWriter schemaChangeEventWriter = new SchemaChangeEventWriter();
+    private static final InsertRowsWriter insertRowsWriter = new InsertRowsWriter();
+    private static final DeleteRowsWriter deleteRowsWriter = new DeleteRowsWriter();
+    private static final UpdateRowsWriter updateRowsWriter = new UpdateRowsWriter();
+
+    static {
+
+        final Set<Relationship> r = new HashSet<>();
+        r.add(REL_SUCCESS);
+        relationships = Collections.unmodifiableSet(r);
+
+        final List<PropertyDescriptor> pds = new ArrayList<>();
+        pds.add(HOSTS);
+        pds.add(DRIVER_NAME);
+        pds.add(DRIVER_LOCATION);
+        pds.add(USERNAME);
+        pds.add(PASSWORD);
+        pds.add(DATABASE_NAME_PATTERN);
+        pds.add(TABLE_NAME_PATTERN);
+        pds.add(CONNECT_TIMEOUT);
+        pds.add(DIST_CACHE_CLIENT);
+        pds.add(RETRIEVE_ALL_RECORDS);
+        pds.add(INIT_SEQUENCE_ID);
+        pds.add(INIT_BINLOG_FILENAME);
+        pds.add(INIT_BINLOG_POSITION);
+        propDescriptors = Collections.unmodifiableList(pds);
+    }
+
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propDescriptors;
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        if (INIT_BINLOG_FILENAME.getName().equals(propertyDescriptorName)) {
+            return INIT_BINLOG_FILENAME;
+        }
+        if (INIT_BINLOG_POSITION.getName().equals(propertyDescriptorName)) {
+            return INIT_BINLOG_POSITION;
+        }
+        return null;
+    }
+
+    @OnScheduled
+    public void setup(ProcessContext context) {
+
+        final ComponentLog logger = getLogger();
+
+        final StateManager stateManager = context.getStateManager();
+        final StateMap stateMap;
+
+        try {
+            stateMap = stateManager.getState(Scope.LOCAL);
+        } catch (final IOException ioe) {
+            logger.error("Failed to retrieve observed maximum values from the State Manager. Will not attempt "
+                    + "connection until this is accomplished.", ioe);
+            context.yield();
+            return;
+        }
+
+        Map<String, String> statePropertiesMap = new HashMap<>(stateMap.toMap());
```

Comment:
I agree, I tried a few different things with respect to state and binlog position, this seemed to work ok but as you point out, it is not the right solution due to initial connection issues. I will make the change, thanks!